core/src/ledger/storage/mod.rs

```diff
@@ -1086,19 +1142,23 @@ where
             if let Some(epoch) = self.block.pred_epochs.get_epoch(min_height) {
                 if epoch.0 == 0 {
                     return Ok(());
-                } else {
-                    // get the start height of the previous epoch because the
-                    // Merkle tree stores at the starting
-                    // height of the epoch would be used
-                    // to restore stores at a height (> min_height) in the epoch
-                    self.db.prune_merkle_tree_stores(
-                        batch,
-                        epoch.prev(),
-                        &self.block.pred_epochs,
-                    )?;
+                }
+                // Remove stores at the previous epoch because the Merkle tree
+                // stores at the starting height of the epoch would be used to
+                // restore stores at a height (> min_height) in the epoch
+                for st in StoreType::iter_provable() {
+                    self.db.prune_merkle_tree_store(batch, st, epoch.prev())?;
                 }
             }
         }
+        // remove non-provable stores at the previous epoch
```

Comment:
we can have an extra condition in `prune_merkle_tree_stores` to remove bridge pool stores with nonces lower than the latest bridge pool nonce signed by a quorum of validators. for more context, see the comment at: https://github.com/anoma/namada/issues/2043#issuecomment-1780567749

something like (pseudo code):
```rs
// remove bridge pool stores with outdated nonces
let signed_nonce = get_bridge_pool_signed_nonce();
let trees = get_stored_bridge_pool_trees();

for t in trees.filter(|t| t.nonce < signed_nonce) {
    /* remove t from storage */
}
```

Reply:
https://github.com/anoma/namada/issues/2043#issuecomment-1783562436